### PR TITLE
add draft brainrender-napari tutorial

### LINKED ISSUE #54: tutorial for brainrender napari usage

### PR DESCRIPTION
**Pull request by alessandrofelder**
## Description

**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
`brainrender-napari` is nearing an initial release, but we have no users docs.

**What does this PR do?**
Add a basic tutorial for how to interact with the BG atlas api via napari.

## References

Closes #54 
Partially addresses https://github.com/brainglobe/brainrender-napari/issues/52

## How has this PR been tested?

Downloaded artefact of website looks reasonable to me.

## Is this a breaking change?

no

## Does this PR require an update to the documentation?

no

## Checklist:

- [n/a] The code has been tested locally
- [n/a] Tests have been added to cover all new functionality (unit & integration)
- [n/a] The documentation has been updated to reflect any changes
- [n/a] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+1. Open `napari`
+1. Install `brainrender-napari` by selecting `Plugins > Install/Uninstall plugins` and searching for `brainrender-napari` in the searchbox. Then click on the "Install" button.
+1. Open the `Brainrender` widget by selecting `Plugins > Brainrender (brainrender-napari)`. 
+    * A widget should appear on the right hand side.
+1. (You can skip this step if you've already used the `mpin_zfish_atlas` in Brainglobe before) In the `Brainrender` widget's "Atlas table view", double-click the 6th row, which contains the `mpin_zfish_1um` atlas. Answer "Yes" to the appearing dialog (this may take a while). 
+    * You have now downloaded the atlas.
+1. In the `Brainrender` widget's "Atlas table view", double-click the 6th row, which contains the `mpin_zfish_1um` atlas (again). 
+    * You have now added the annotations image and the reference image to napari.
+    * A "Structures" widget appears below the existing widget.
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:13]: I find this slightly hard to parse, it says to skip this step (double click), but then the next step is to double click. I find having a shorter list, then a separate box explaining the download easier to parse (not sure if you agree). I also removed reference to the 6th row, as that is likely to change in the future. 

```suggestion
1. In the `Brainrender` widget's "Atlas table view", double-click the row which contains the `mpin_zfish_1um` atlas
    * You have now added the annotations image and the reference image to napari.
    * A "Structures" widget appears below the existing widget.

:::{note}
If you haven't downloaded the atlas before, the plugin will prompt you to download it (click yes). It may take a long time
(depending on your internet speed) so please be patient. Once downloaded, double clicks the atlas to view.
:::
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+1. Toggle the visualisation to a 3D view by pressing `Ctrl+Y` (`Cmd+Y` on Mac) on your keyboard.
+1. Navigate the structures tree in the "Structures" widget by opening "forebrain". Double-click on "telencephalon".
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:15]: I don't think "Structures widget" is clear for many users. It doesn't say Structures anywhere, and they probably don't know what a widget is.

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    * You have now added a mesh layer of a brain structure to napari.
+1. Back in the "Atlas table view", right-click on the 6th row. In the appearing context menu, select "GAD1b"
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:17]:
```suggestion
1. Back in the "Atlas table view", right-click on the `mpin_zfish_1um` row. In the menu that appears, select `GAD1b`
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    * You have now added an additional reference image to napari
+
+You have now added all possible types of brainglobe atlas parts (annotations image, reference image, structure mesh, additional reference) to napari - well done! You can now add any of the other atlases listed in the "Atlas table view" if you like (note that not all atlases have additional references!). Hover over any of the elements in the `Brainrender` widget to get additional hints about how to use them!
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:20]:
```suggestion
You have now added all possible types of BrainGlobe atlas parts (annotations image, reference image, structure mesh, additional reference) to napari - well done! You can now add any of the other atlases listed in the "Atlas table view" if you like (note that not all atlases have additional references!). Hover over any of the elements in the `brainrender` widget to get additional hints about how to use them!
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+1. Open `napari`
+1. Install `brainrender-napari` by selecting `Plugins > Install/Uninstall plugins` and searching for `brainrender-napari` in the searchbox. Then click on the "Install" button.
+1. Open the `Brainrender` widget by selecting `Plugins > Brainrender (brainrender-napari)`. 
+    * A widget should appear on the right hand side.
+1. (You can skip this step if you've already used the `mpin_zfish_atlas` in Brainglobe before) In the `Brainrender` widget's "Atlas table view", double-click the 6th row, which contains the `mpin_zfish_1um` atlas. Answer "Yes" to the appearing dialog (this may take a while). 
+    * You have now downloaded the atlas.
+1. In the `Brainrender` widget's "Atlas table view", double-click the 6th row, which contains the `mpin_zfish_1um` atlas (again). 
+    * You have now added the annotations image and the reference image to napari.
+    * A "Structures" widget appears below the existing widget.
+1. Toggle the visualisation to a 3D view by pressing `Ctrl+Y` (`Cmd+Y` on Mac) on your keyboard.
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:14]: Could you add details about the button to click, and then say "alternatively press ....". I think button clicking is more intuitive for many users.

**Review by adamltyson — APPROVED** [review]
I've added some very nitpicky suggestions. I also think my suggestion for the admonition box broke the sequential numbering. Does putting everything as `1.` automatically create the numbering? I've never seen this.

Generally it would be cool to add three things:
- Some images showing the buttons to click, and what a successful outcome is. 
- Some more details as to why. Things like "the default reference image is X, if you want to view the other reference image Y, do this..."
- A (short) docs page alongside the other tools, so we can continue the pattern of comprehensive docs, and a specific application tutorial.

File: docs/source/tutorials/visualise-atlas-napari.md (hunk 2 above)
Comment (alessandrofelder) [reply docs/source/tutorials/visualise-atlas-napari.md:18]: It does say "Structures" now that 
I've basically adapted terminology from `bg-atlasapi` but happy change the user-facing part to something better?
What would a better term be? "Regions"? "Subvolumes"? "Meshes"?

**Comment by alessandrofelder** [comment]
> * Some more details as to why. Things like "the default reference image is X, if you want to view the other reference image Y, do this..."

While I'm happy not to be too dogmatic about it, I would argue that this goes [against what the diataxis framework suggests for tutorials](https://diataxis.fr/tutorials/#offer-only-minimum-necessary-explanation)
>It can seem problematic that we are asking a user to do things, without much explanation why. In practice, for the learner, it rarely is. The learner is focused on following your directions and getting a result; their time for wanting to know more about the why of what they’re doing will come later.

**Comment by alessandrofelder** [comment]
> A (short) docs page alongside the other tools, so we can continue the pattern of comprehensive docs, and a specific application tutorial.
Yes, good call - separate tickets opened (#56, #57)

File: docs/source/tutorials/images/visualise-atlas-napari.png
[the remark this answers is not in the document]
Comment (alessandrofelder) [reply docs/source/tutorials/images/visualise-atlas-napari.png:1]: Done :heavy_check_mark:

File: docs/source/tutorials/visualise-atlas-napari.md (hunk 2 above)
Comment (adamltyson) [reply docs/source/tutorials/visualise-atlas-napari.md]: Maybe `3D brain region meshes` to be super clear?

**Comment by adamltyson** [comment]
> > * Some more details as to why. Things like "the default reference image is X, if you want to view the other reference image Y, do this..."
> 
> While I'm happy not to be too dogmatic about it, I would argue that this goes [against what the diataxis framework suggests for tutorials](https://diataxis.fr/tutorials/#offer-only-minimum-necessary-explanation)
> 
> > It can seem problematic that we are asking a user to do things, without much explanation why. In practice, for the learner, it rarely is. The learner is focused on following your directions and getting a result; their time for wanting to know more about the why of what they’re doing will come later.

Is this contrary to the diataxis framework? I'm not arguing we give them technical details that they don't need to carry out a task, I'm arguing that we should let them know _what_ they're doing. If they aren't a zebrafish biologist, would they learn how to click a button, but not what the button does?

File: docs/source/tutorials/visualise-atlas-napari.md (hunk 2 above)
Comment (alessandrofelder) [reply docs/source/tutorials/visualise-atlas-napari.md]: Cool - to be addressed by https://github.com/brainglobe/brainrender-napari/pull/82

File: docs/source/tutorials/visualise-atlas-napari.md (hunk 5 above)
Comment (alessandrofelder) [reply docs/source/tutorials/visualise-atlas-napari.md]: I've switched to an entirely mouse-clickey way of doing this, with a picture.

**Comment by alessandrofelder** [comment]
Thanks a lot for the thorough review, @adamltyson - maybe have another looks as I've changed quite a bit.

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+## Visualise an atlas in napari
+
+BrainGlobe atlases are made of different kinds of parts. These include the annotations image, the (default) reference image, 3D brain region meshes and, optionally, additional reference images. In this tutorial, you will use `brainrender-napari` to visualise each of these parts in the 3D viewer napari through a series of simple clicks. The [`mpin_zfish_1um` zebrafish brain atlas](https://doi.org/10.1016/j.neuron.2019.04.034) will serve as an example.
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:3]:
```suggestion
BrainGlobe atlases are made of different components. These include the annotations image, the (default) reference image, 3D brain region meshes and, optionally, additional reference images. In this tutorial, you will use `brainrender-napari` to visualise each of these components in the 3D viewer napari through a series of simple clicks. The [`mpin_zfish_1um` zebrafish brain atlas](https://doi.org/10.1016/j.neuron.2019.04.034) will serve as an example. 
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+## Visualise an atlas in napari
+
+BrainGlobe atlases are made of different kinds of parts. These include the annotations image, the (default) reference image, 3D brain region meshes and, optionally, additional reference images. In this tutorial, you will use `brainrender-napari` to visualise each of these parts in the 3D viewer napari through a series of simple clicks. The [`mpin_zfish_1um` zebrafish brain atlas](https://doi.org/10.1016/j.neuron.2019.04.034) will serve as an example. 
+
+:::{note}
+You will need `napari` installed on your computer - please follow [`napari`'s installation instructions to do so](https://napari.org/stable/tutorials/fundamentals/installation.html).
+:::
+
+1. Open `napari`
+2. Install `brainrender-napari` by selecting `Plugins > Install/Uninstall plugins` and searching for `brainrender-napari` in the searchbox. Then click on the "Install" button.
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:10]:
```suggestion
2. Install `brainrender-napari` by selecting `Plugins > Install/Uninstall plugins` and searching for `brainrender-napari` in the searchbox. Then click on the `Install` button.
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+3. Open the `brainrender` widget by selecting `Plugins > Brainrender (brainrender-napari)` in the napari menu bar near the top left of the window. 
+![brainrender widget](./images/brainrender-napari/plugin-menu-brainrender-napari.png)
+**The brainrender widget appears on the right hand side of the window.**
+
+4. In the `Brainrender` widget's "Atlas table view" section, double-click the row which contains the `mpin_zfish_1um` atlas
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:15]:
```suggestion
4. In the `brainrender` widget's `Atlas table view` section, double-click the row which contains the `mpin_zfish_1um` atlas.
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+![brainrender widget with added annotations](./images/brainrender-napari/added-brainrender-napari.png)
+**You have now added the annotations image and the default reference image to napari: They appear as layers in the napari layers list on the lower left of the window. A "3D Brain region meshes" section appears below the "Atlas table view" section.**
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:17]:
```suggestion
**You have now added the annotations image and the default reference image to napari: They appear as layers in the napari layers list on the lower left of the window. A `3D Brain region meshes` section appears below the `Atlas table view` section.**
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+:::{note}
+If you haven't downloaded the atlas before, the plugin will prompt you to download it (click yes). It may take a long time
+(depending on your internet speed) so please be patient. Once downloaded, double click again to view the atlas.
+:::
+
+5. Toggle the napari display from 2D to 3D by pressing the button with the square icon on the lower left of the window.
+![brainrender widget with 3d display](./images/brainrender-napari/toggle-ndisplay-brainrender-napari.png)
+**The annotations image should now be displayed in 3D.**
+6. Navigate the brain region tree in the "3D Brain region meshes" section by opening "forebrain". Double-click on "telencephalon".
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:27]:
```suggestion
6. Navigate the brain region tree in the `3D Brain region meshes` section by opening "forebrain". Double-click on `telencephalon`.
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+![brainrender widget with region mesh](./images/brainrender-napari/add-region-brainrender-napari.png)
+ **You have now added a 3D brain region mesh layer, which appears as a mesh in the viewer and as a new layer in the layers list**
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:29]:
```suggestion
 **You have now added a 3D brain region mesh layer, which appears as a mesh in the viewer and as a new layer in the layers list.**
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 12]
```diff
@@ -0,0 +1,34 @@
+## Visualise an atlas in napari
+
+BrainGlobe atlases are made of different kinds of parts. These include the annotations image, the (default) reference image, 3D brain region meshes and, optionally, additional reference images. In this tutorial, you will use `brainrender-napari` to visualise each of these parts in the 3D viewer napari through a series of simple clicks. The [`mpin_zfish_1um` zebrafish brain atlas](https://doi.org/10.1016/j.neuron.2019.04.034) will serve as an example. 
+
+:::{note}
+You will need `napari` installed on your computer - please follow [`napari`'s installation instructions to do so](https://napari.org/stable/tutorials/fundamentals/installation.html).
+:::
+
+1. Open `napari`
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:9]:
```suggestion
1. Open `napari`.
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+(depending on your internet speed) so please be patient. Once downloaded, double click again to view the atlas.
+:::
+
+5. Toggle the napari display from 2D to 3D by pressing the button with the square icon on the lower left of the window.
+![brainrender widget with 3d display](./images/brainrender-napari/toggle-ndisplay-brainrender-napari.png)
+**The annotations image should now be displayed in 3D.**
+6. Navigate the brain region tree in the "3D Brain region meshes" section by opening "forebrain". Double-click on "telencephalon".
+![brainrender widget with region mesh](./images/brainrender-napari/add-region-brainrender-napari.png)
+ **You have now added a 3D brain region mesh layer, which appears as a mesh in the viewer and as a new layer in the layers list**
+7. Back in the "Atlas table view" section, right-click on the `mpin_zfish_1um` row. In the menu that appears, select `GAD1b`
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:30]:
```suggestion
7. Back in the "Atlas table view" section, right-click on the `mpin_zfish_1um` row. In the menu that appears, select `GAD1b`.
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+![brainrender widget with additional reference](./images/brainrender-napari/additional-reference-brainrender-napari.png)
+**You have now added an additional reference image, which appears as a grey scale image in the viewer and as a new layer in the layers list**
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:32]:
```suggestion
**You have now added an additional reference image, which appears as a grey scale image in the viewer and as a new layer in the layers list.**
```

File: docs/source/tutorials/visualise-atlas-napari.md [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+You have now added all possible kinds of BrainGlobe atlas parts (annotations image, reference image, 3D brain region mesh, additional reference) to napari - well done! You can now add any of the other atlases listed in the "Atlas table view" if you like (note that not all atlases have additional references!). Hover over any of the elements in the `brainrender` widget to get additional hints about how to use them!
```
Comment (adamltyson) [inline docs/source/tutorials/visualise-atlas-napari.md:34]:
```suggestion
You have now added all possible kinds of BrainGlobe atlas components (annotations image, reference image, 3D brain region mesh, additional reference) to napari - well done! You can now add any of the other atlases listed in the `Atlas table view` if you like (note that not all atlases have additional references!). 

:::{note}
Hover over any of the elements in the `brainrender` widget to get additional hints about how to use them!
:::
```

**Review by adamltyson — APPROVED** [review]
Very nice. I left some v small stylistic suggestions.
